plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchColumnHandle.java

```diff
@@ -13,28 +13,45 @@
  */
 package io.trino.plugin.opensearch;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
 import io.trino.spi.connector.ColumnHandle;
 import io.trino.spi.type.Type;
 
+import java.util.List;
+
 import static java.util.Objects.requireNonNull;
 
 public record OpenSearchColumnHandle(
-        String name,
+        String baseName,
+        List<String> dereferencePath,
         Type type,
         DecoderDescriptor decoderDescriptor,
         boolean supportsPredicates)
         implements ColumnHandle
 {
     public OpenSearchColumnHandle
     {
-        requireNonNull(name, "name is null");
+        requireNonNull(baseName, "baseName is null");
+        dereferencePath = ImmutableList.copyOf(requireNonNull(dereferencePath, "dereferencePath is null"));
         requireNonNull(type, "type is null");
         requireNonNull(decoderDescriptor, "decoderDescriptor is null");
     }
 
+    @JsonIgnore
+    public String getQualifiedName()
+    {
+        return Joiner.on('.')
```

Comment:
This can be problematic, e.g., if the field names contain `.`. There's no need to turn the reference chain into a String.

Reply:
Discussed offline that in dot corresponds to object type in  os. There is an assertion showing the same behaviour https://github.com/trinodb/trino/pull/22646/commits/9a84ed0d3971a6b91387a3ebc2be7589225fceb9#diff-4abea133110790acaf59c5c1e2046428a8d761523cab0d8639c3532a8bfdd7a0R1045-R1046